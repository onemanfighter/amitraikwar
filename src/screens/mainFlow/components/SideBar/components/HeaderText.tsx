import { Button } from '@chakra-ui/react';
import { useColorSelector } from '@components';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';

const HeaderText = () => {
  const { t } = useTranslation();
  const { text } = useColorSelector();

  return (
    <Button
      fontSize={'2xl'}
      as={Link}
      to={'/'}
      bgGradient={text.Hero}
      bgClip={'text'}
      fontWeight={'800'}
      fontFamily={'monospace'}
      variant={'ghost'}
      margin={'4'}
      padding={'6'}
      _hover={{
        bg: 'green.500',
        color: 'white',
        boxShadow: 'lg',
      }}
      transition={'all 0.3s'}
    >
      {t('MainHomeScreen.navigation.title')}
    </Button>
  );
};

export default HeaderText;
