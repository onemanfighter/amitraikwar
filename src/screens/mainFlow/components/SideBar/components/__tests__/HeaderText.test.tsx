import { render, screen } from '@testing-library/react';
import HeaderText from '../HeaderText';
import { ThemeProvider } from '@providers';
import { CSSReset } from '@chakra-ui/react';
import { BrowserRouter } from 'react-router-dom';

describe('Header Text', () => {
  it('should renders correctly', () => {
    const { container } = render(
      <ThemeProvider>
        <BrowserRouter>
          <HeaderText />
        </BrowserRouter>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render header text correctly', () => {
    render(
      <ThemeProvider>
        <BrowserRouter>
          <HeaderText />
        </BrowserRouter>
      </ThemeProvider>,
    );

    expect(screen.findByText('Amit Raikwar')).toBeDefined();
  });
});
