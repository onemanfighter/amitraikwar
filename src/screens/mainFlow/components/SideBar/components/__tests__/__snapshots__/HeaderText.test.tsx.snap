// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Text should renders correctly 1`] = `
<div>
  <script
    id="chakra-script"
  >
    !(function(){try{var a=function(c){var v="(prefers-color-scheme: dark)",h=window.matchMedia(v).matches?"dark":"light",r=c==="system"?h:c,o=document.documentElement,s=document.body,l="chakra-ui-light",d="chakra-ui-dark",i=r==="dark";return s.classList.add(i?d:l),s.classList.remove(i?l:d),o.style.colorScheme=r,o.dataset.theme=r,r},n=a,m="light",e="chakra-ui-color-mode",t=localStorage.getItem(e);t?a(t):localStorage.setItem(e,a(m))}catch(a){}})();
  </script>
  <a
    class="chakra-button css-1ozkr8m"
    href="/"
  >
    Amit Raikwar
  </a>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
